app/dependencies.php

```diff
@@ -151,11 +151,38 @@ function (ContainerInterface $c): Auth\DonationPublicAuthMiddleware {
             return new Client\Mailer($settings, $c->get(LoggerInterface::class));
         },
 
+        \Symfony\Component\Clock\ClockInterface::class =>
+            function (ContainerInterface $_c): \Symfony\Component\Clock\ClockInterface {
+                return new \Symfony\Component\Clock\Clock();
```

Comment:
Introducing the [Symofony Clock](https://symfony.com/doc/current/components/clock.html) component here. @NoelLH @dorota-joanna what do you think of making it a rule from now on to use the clock component rather than direct access to PHP functions when we need to deal with the system clock, at least when what it does with that clock is something we care to test? 

IMHO it's often hard to write and read tests for things that directly uses the system clock, so injecting a clock a dependency is a god solution.

Reply:
Sounds good!